debian_specs/passenger/passenger_free_ruby.c

```diff
@@ -15,22 +15,15 @@
 int
 main(int argc, const char *argv[]) {
 	try_exec("ruby", argc, argv);
-	try_exec("/usr/bin/ruby3.4", argc, argv);
-	try_exec("/usr/bin/ruby3.3", argc, argv);
-	try_exec("/usr/bin/ruby3.2", argc, argv);
-	try_exec("/usr/bin/ruby3.1", argc, argv);
-	try_exec("/usr/bin/ruby3.0", argc, argv);
-	try_exec("/usr/bin/ruby2.7", argc, argv);
-	try_exec("/usr/bin/ruby2.6", argc, argv);
-	try_exec("/usr/bin/ruby2.5", argc, argv);
-	try_exec("/usr/bin/ruby2.4", argc, argv);
-	try_exec("/usr/bin/ruby2.3", argc, argv);
-	try_exec("/usr/bin/ruby2.2", argc, argv);
-	try_exec("/usr/bin/ruby2.1", argc, argv);
-	try_exec("/usr/bin/ruby2.0", argc, argv);
+	char* exe = "/usr/bin/rubyX.XX";
```

Comment:
This actually won't work on all platforms. A static string literal is, on some platforms, read-only memory.

Better to turn it into a char array so that it's writable memory:

```
char exe[] = "/usr/bin/rubyX.XX";
```

Also, instead of passing 18 to snprintf, use `sizeof(exe)` to avoid mistakes.